### ISSUE TITLE
Broken folding when using LF instead of CRLF

### ISSUE THREAD

**Issue by pvandam** [notepad-plus-plus/notepad-plus-plus#350]
For a user defined language with parentheses and brackets as folders.

No problem with CRLF:
![image](https://cloud.githubusercontent.com/assets/6884673/8392764/97495878-1cf5-11e5-83bf-3e631e352501.png)

Switch to LF and see:
![image](https://cloud.githubusercontent.com/assets/6884673/8392766/ac50cb48-1cf5-11e5-88ba-64f6c0de73f7.png)

switching EOL on the text below will trigger this behavior.

```
    "token" (
        "token" (
            "token" -> [
                (
                    "token" (
                        "version" -> "a4afdc5"
                    )
                )
            ]
        )
    )
    view "token" widget "token" : "token" (
        "token" -> state "token" ( )
    )
```

version 6.7.9.2 on Windows 7 x64.

**Comment by Rikk** [comment]
Bug in the custom language?

**Comment by pvandam** [comment]
It is a user defined language. I would think line ending/folding bugs could
not be introduced by those.

**Comment by Cryosaur** [comment]
I also just noticed this issue with an older version and see that it has not yet been resolved in 6.7.9.2.

My user-defined language has keywords for folding (using style 2).  If a file has CRLF line endings, folding works perfectly; when I switch to one of the other EOL formats, folding immediately becomes wonky, moving the start and/or end fold points a line or more below where they should be, accuracy decreasing with the number of fold points.

As a quick test, I tried the same thing on a C# file and this folding problem did not occur.  So maybe it is a bug related to user-defined languages, though I am confident that it is not pilot error.

**Comment by Combinatix** [comment]
Yes. I'm having the same problem. I've made very basic example of a code to demonstrate this issue. Attached is the example of code and also UDL language. The folding does not work properly when UNIX EOL

[UDL Folding Fail.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/266761/UDL.Folding.Fail.zip)

NPP6.9.1, UDL 2.1.0.12

**Comment by Fiooodooor** [comment]
It's still and open issue. Folding just won't work with LF only.

**Comment by donho** [comment]
Update your Notepad++ to v7.6.6.
If this bug persists in the new release, please open a new issue with the debug information.

**Comment by bReathl3sS** [comment]
Still not fixed, see #5499
